moto/apigateway/models.py

```diff
@@ -754,6 +765,86 @@ def get_rest_api(self, function_id):
             raise RestAPINotFound()
         return rest_api
 
+    def put_rest_api(self, function_id, api_doc, mode="merge", fail_on_warnings=False):
+        if mode not in ["merge", "overwrite"]:
+            raise InvalidOpenApiModeException()
+
+        if api_doc.get("swagger") is not None or (
+            api_doc.get("openapi") is not None and api_doc["openapi"][0] != "3"
+        ):
+            raise InvalidOpenApiDocVersionException()
+
+        if mode == "overwrite":
+            self.resources = {}
+            self.add_child("/")  # Add default child
+
+        try:
+            if fail_on_warnings:
+                validate_spec(api_doc)
+            for (path, resource_doc) in sorted(
+                api_doc["paths"].items(), key=lambda x: x[0]
+            ):
+                parent_path_part = path[0 : path.rfind("/")] or "/"
+                parent_resource_id = self.apis[function_id].get_resource_for_path(
+                    parent_path_part
+                )
+                resource = self.create_resource(
+                    function_id=function_id,
+                    parent_resource_id=parent_resource_id,
+                    path_part=path[: path.rfind("/")],
+                )
+
+                for (method_type, method_doc) in resource_doc.items():
+                    if method_doc.get("x-amazon-apigateway-integration") is None:
+                        self.create_method(function_id, resource.id, method_type, None)
+                        for (response_code, response_doc) in method_doc.get(
+                            "responses", {}
+                        ).items():
+                            self.create_method_response(
+                                function_id, resource.id, method_type, response_code
+                            )
+                    else:
+                        self.create_integration(
+                            function_id=function_id,
+                            resource_id=resource.id,
+                            method_type=method_type,
+                            integration_type=method_doc[
```

Comment:
This becomes a bit difficult to read. Might be better to assign temp variables first, before calling the `create_integration`-function
(I.e. `int_type = method_doc[".."]["type"]`, etc)